src/cljs/main/broadfcui/common/method/config_io.cljs

```diff
@@ -57,11 +64,33 @@
    (fn [{:keys [props state]}]
      (swap! state utils/deep-merge (:values props)))
    :render
-   (fn [{:keys [props this]}]
-     (let [{:keys [default-hidden? entity-type? style can-edit?]} props
+   (fn [{:keys [props state this]}]
+     (let [{:keys [default-hidden? entity-type? style can-edit? begin-editing]} props
+           {:keys [editing?]} @state
            id (gensym "io-table-")]
        [:div {:id id :style style}
+        (when (:show-upload? @state)
+          [modals/OKCancelForm
+           {:header "Upload JSON" :show-cancel? true :cancel-text "Close"
+            :dismiss #(swap! state dissoc :show-upload?)
+            :content [:div {:style {:width 720 :backgroundColor "white" :padding "1em"}}
+                      [import-data/Page
+                       {:on-upload
+                        (fn [{:keys [file-contents]}]
+                          (let [uploaded-inputs (utils/parse-json-string file-contents true)
+                                new-inputs (merge (:inputs @state) uploaded-inputs)]
+                            (when-not editing? (begin-editing))
+                            (swap! state assoc :inputs new-inputs)
+                            (swap! state dissoc :show-upload?)))}]]}])
         [Collapse {:title "Inputs"
+                   :secondary-title (when can-edit?
+                                     [:div {}
```

Comment:
nit picky - `[:div {}` shouldn't be on a line by itself - I would move `(links/create-internal` up